File: gitdot-web/packages/api/src/endpoint/review/merge-diff.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const MergeDiffRequest = z.object({});
export type MergeDiffRequest = z.infer<typeof MergeDiffRequest>;

export const MergeDiffResponse = ReviewResource;
export type MergeDiffResponse = z.infer<typeof MergeDiffResponse>;

export const MergeDiff = {
  path: "/repository/{owner}/{repo}/review/{number}/diff/{position}/merge",
  method: "POST",
  request: MergeDiffRequest,
  response: MergeDiffResponse,
} as const satisfies Endpoint;
export type MergeDiff = typeof MergeDiff;

File: gitdot-api/src/endpoint/review/merge_diff.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewResource};

pub struct MergeDiff;

impl Endpoint for MergeDiff {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/diff/{position}/merge";
    const METHOD: http::Method = http::Method::POST;

    type Request = MergeDiffRequest;
    type Response = MergeDiffResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct MergeDiffRequest {}

pub type MergeDiffResponse = ReviewResource;
